### PR TITLE
Add support for external logger

### PR DESCRIPTION
**Pull request by jsumners**
This PR adds direct support for using an external logger. Without this PR you must pass an external logger in like so: `const fastify = require('fastify')({logger: {logger: externalLogger}})`. Additionally, since this is merely directly passing the logger into `pino-http` there is no guarantee it will have the methods that `fastify` needs. This PR verifies the logger prior to attempting to use it.

### CODE REVIEW

**Comment by jsumners** [comment]
I don't know what Travis's problem is. The tests are working just fine with v4.8.0 on my system.

File: test/logger.test.js [hunk 1]
```diff
@@ -42,3 +42,43 @@ test('test log stream', t => {
...
+
+  const logger = require('pino')(splitStream)
+  logger.info('log success')
+
+  const localFastify = Fastify({logger: logger})
+
+  localFastify.get('/foo', function (req, reply) {
+    t.ok(req.log)
+    req.log.info('log success')
+    reply.send(null, 200, { hello: 'world' })
```
Comment (delvedor) [inline test/logger.test.js]: For some reason we didn't update this file to the new reply syntax, can you do it?
old: `reply(null, 200, { hello: 'world' })`
new: `reply.send({ hello: 'world' })`

Comment (delvedor) [inline test/logger.test.js]: For some reason we didn't update this file to the new *reply* syntax, can you do it? :)
old: `reply(null, 200, { hello: 'world' })`
new: `reply.send({ hello: 'world' })`

File: fastify.js [hunk 2]
```diff
@@ -15,15 +15,32 @@ const buildSchema = require('./lib/validation').build
 const buildNode = require('./lib/tier-node')
 const hooksManager = require('./lib/hooks')
 
+function isValidLogger (logger) {
```
Comment (delvedor) [inline fastify.js]: Maybe move this function inside `lib/validation.js`?
cc @mcollina

**Pull request merged**